File: src/main/java/com/d20charactersheet/framework/dac/dao/sql/rowmapper/WeaponFamilyRowMapper.java
package com.d20charactersheet.framework.dac.dao.sql.rowmapper;

import java.sql.SQLException;

import org.jetbrains.annotations.NotNull;

import com.d20charactersheet.framework.boc.model.WeaponFamily;
import com.d20charactersheet.framework.dac.dao.sql.DataRow;

/**
 * Maps a data row to a new WeaponFamily instance.
 */
public class WeaponFamilyRowMapper extends BaseRowMapper {

    @NotNull
    @Override
    public Object mapRow(@NotNull DataRow dataRow) throws SQLException {
        final WeaponFamily weaponFamily = new WeaponFamily();
        weaponFamily.setId(dataRow.getInt(0));
        weaponFamily.setName(dataRow.getString(1));
        return weaponFamily;
    }

}

File: src/main/java/com/d20charactersheet/framework/dac/dao/sql/rowmapper/WeaponFamilyRowMapper.kt
package com.d20charactersheet.framework.dac.dao.sql.rowmapper

import com.d20charactersheet.framework.boc.model.WeaponFamily
import com.d20charactersheet.framework.dac.dao.sql.DataRow
import java.sql.SQLException

class WeaponFamilyRowMapper : BaseRowMapper() {

    /**
     * Maps a data row to a new WeaponFamily instance.
     */
    @Throws(SQLException::class)
    override fun mapRow(dataRow: DataRow): Any {
        val weaponFamily = WeaponFamily()
        weaponFamily.id = dataRow.getInt(0)
        weaponFamily.name = dataRow.getString(1)
        return weaponFamily
    }
}